### PR TITLE
Merge mkdocs into branch 0.3

### PR DESCRIPTION
**Pull request by farodin91**
Part of Issue #1437 
Preview: https://farodin91.github.io/janusgraph/mkdocs03/

To compare: https://github.com/JanusGraph/janusgraph/compare/janusgraph:v0.2...farodin91:mkdocs03

Following changes:
* .travis.yml
* mkdocs.yml
* a lot of markdown files
* docs.Dockerfile
* github template

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [x] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?
- [x] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by farodin91** [comment]
@mbrukman Do know what the problem is why the CLA fails?

**Comment by farodin91** [comment]
@mbrukman Do know what the problem is why the CLA fails?

**Comment by porunov** [comment]
1) https://farodin91.github.io/janusgraph/index-backend/field-mapping/
Section `Custom parameters` is available only from `0.3.1`.

2) https://farodin91.github.io/janusgraph/advanced-topics/serializer/
Section `Datatype and Attribute Serializer Configuration`. ObjectNode and ArrayNode are available only from `0.3.2`.

File: docs/advanced-topics/serializer.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 Datatype and Attribute Serializer Configuration
 ===============================================
 
+!!! Info
+    Datatype and Attribute Serializer Configuration are added in JanusGraph version `0.3.2`
```
Comment (porunov) [inline docs/advanced-topics/serializer.md]: No, serialization was already implemented. I mean only `ObjectNode` and `ArrayNode`.
I.e. 
`ObjectNode` and `ArrayNode` serializers are added in JanusGraph version `0.3.2`

**Comment by porunov** [comment]
@farodin91 Could you please update gh-pages so that demo website would be up to date?

**Comment by farodin91** [comment]
Is updated, if you use the link in the pr. Otherwise, i have to do each time multiple merges.

**Comment by farodin91** [comment]
@porunov Could you try to activate github pages for this repo? It should be possbile using the gh-pages branch. https://janusgraph.github.io/janusgraph/ 

Aftwerward, I push a v0.3 branch and merge it into the v0.3. Than, we have to go fast to not stop the development of JanusGraph. 
ToDo:
* Delete old branches after merging mkdocs, for example 0.2, 0.3, 0.4
* Do such a pr for ` v0.4`, `master`
* Finally, use the doc as default documenation

**Comment by porunov** [comment]
@farodin91 Do I need to enable gh-pages for `https://github.com/JanusGraph/docs.janusgraph.org` or `https://github.com/JanusGraph/janusgraph`?

**Comment by farodin91** [comment]
> Do I need to enable gh-pages for `https://github.com/JanusGraph/docs.janusgraph.org` or `https://github.com/JanusGraph/janusgraph`?

For janusgraph/janusgraph.

We can do a hangout tomorrow to get mkdocs merge through.

**Comment by porunov** [comment]
>> For janusgraph/janusgraph.

Will it have `docs.janusgraph.org` for janusgraph docs?

>> We can do a hangout tomorrow to get mkdocs merge through.

Sure. Just send me email establish the time

**Comment by farodin91** [comment]
> Will it have docs.janusgraph.org for janusgraph docs?

To get the new documentation ready, we can use https://janusgraph.github.io/janusgraph/ which is default route for this repo. Afterward, we can update the dns for the url or move the (mk)docs into the docs.janusgraph.org repo.

**Comment by porunov** [comment]
@farodin91 Looks like gh-pages are enabled for janusgraph

**Comment by farodin91** [comment]
We should be able to access following site if it is enabled: https://janusgraph.github.io/janusgraph/v0.2/index.html

**Comment by farodin91** [comment]
In my repo it looks so:
image

**Comment by porunov** [comment]
I JanusGraph repo it looks like:
Screenshot from 2019-08-22 10-14-46

**Comment by farodin91** [comment]
Could we switch the source to something different than back to gh-pages.

**Comment by porunov** [comment]
@farodin91 Done. Looks like it works now: https://janusgraph.github.io/janusgraph/v0.2/

**Comment by farodin91** [comment]
@porunov  I will prepare a branch for 0.4 and on monday, i have some time to connect using hangout or zoom.

**Comment by porunov** [comment]
@farodin91 Sure. Let me know what you decide.
Also, should we merge this and #1756 now or on Monday? Both PRs looks good to me

**Comment by farodin91** [comment]
@porunov If you time now, we can do it. You should got a mail.

Todo:
I will push a v0.3 and v0.4 branch.

**Pull request merged**